src/rust/engine/src/externs/engine_aware.rs

```diff
@@ -10,20 +10,10 @@ use crate::Types;
 use cpython::{PyDict, PyString, Python};
```

Comment:
Could this be a pre-factor PR? Seems unrelated to rest of the changes fwict, and the main change is already intricate

Reply:
It could be, but then both of them would need to be cherry-picked, which would be a bit of a pain.

The change in this file is just to remove a `trait` which wasn't actually used for anything, but which made a the `Types` parameter necessary for every `resolve` method, even though it wasn't used for most of them.